src/util/make-periodic-task.ts

```diff
@@ -12,41 +22,38 @@ export interface PeriodicTask {
 export function makePeriodicTask(
   task: () => Promise<void> | void,
   msGap: number,
-  opts: {
-    onError?: (error: any) => void
-  } = {}
+  opts: PeriodicTaskOptions = {}
 ): PeriodicTask {
-  const {
-    onError = (e: any) => {
-      // do nothing
-    }
-  } = opts
+  const { onError = () => undefined } = opts
 
   // A started task will keep bouncing between running & waiting.
   // The `running` flag will be true in the running state,
   // and `timeout` will have a value in the waiting state.
   let running = false
-  let timeout: any
+  let timeout: ReturnType<typeof setTimeout> | undefined
 
-  function run(): void {
+  function startRunning(): void {
     timeout = undefined
     if (!out.started) return
     running = true
-    new Promise(resolve => resolve(task())).catch(onError).then(wait, wait)
+    new Promise(resolve => resolve(task()))
+      .catch(onError)
+      .then(startWaiting, startWaiting)
```

Comment:
`.finally(startWaiting)`?

Reply:
The `.finally` method isn't supported on all platforms the core needs to run on, and since is just a sync of the core's changes into this library, it wouldn't make sense to use that.